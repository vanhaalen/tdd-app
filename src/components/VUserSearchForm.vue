<template>
  <form @submit.prevent="$emit('submitted', username)">
    <input type="text" v-model="username" />
    <button type="submit">Submit</button>
  </form>
</template>

<script>
export default {
  name: "VUserSearchForm",
  data() {
    return {
      username: ""
    };
  }
};
</script>

<style scoped></style>
