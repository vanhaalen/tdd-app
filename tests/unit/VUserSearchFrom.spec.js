import { shallowMount } from "@vue/test-utils";
import VUserSearchForm from "@/components/VUserSearchForm";

describe("VUserSearchForm", () => {
  const build = () => {
    const wrapper = shallowMount(VUserSearchForm, {
      attachToDocument: true
    });

    return {
      wrapper,
      input: () => wrapper.find("input"),
      button: () => wrapper.find("button")
    };
  };

  it("should render the component", () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render the main child components", () => {
    // arrange
    const { input, button } = build();

    // assert
    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it("should emit 'submitted' event when submitting the form", () => {
    // arrange
    const expectedUser = "kuroski";
    const { wrapper, input, button } = build();
    input().element.value = expectedUser;

    // act
    input().trigger("input");
    button().trigger("click");
    // button().trigger("submit");

    // assert
    expect(wrapper.emitted().submitted[0]).toEqual([expectedUser]);
  });
});
