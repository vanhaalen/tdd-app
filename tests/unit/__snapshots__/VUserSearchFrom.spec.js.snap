// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm should render the component 1`] = `<form><input type="text"> <button type="submit">Submit</button></form>`;
